core/database/api/task_router.js

```diff
@@ -0,0 +1,296 @@
+.queryParam('task_id', joi.string().required(), "Task ID")
+.summary('View an existing task record')
+.description('View an existing task record.');
+
+
+router.get('/run', function (req, res) {
+    var task, run_func;
+
+    //console.log("task/run - trans 1");
+
+    try {
+        g_db._executeTransaction({
+            collections: { read: [], write: ["task"] },
+            lockTimeout: 0,
+            waitForSync: true,
+            action: function() {
+                if ( !g_db.task.exists( req.queryParams.task_id ))
+                    throw [ g_lib.ERR_INVALID_PARAM, "Task " + req.queryParams.task_id + " does not exist." ];
+
+                task = g_db.task.document( req.queryParams.task_id );
+                run_func = g_tasks.taskGetRunFunc( task );
+
+                // If the last step is about to run, add exclusive lock, block access to transaction
+                //if ( req.queryParams.step != undefined && req.queryParams.step == task.steps - 2 ){
+                //    exc = ["lock","block"];
+                //}
+
+                // There should never be a wr-wr conflict here b/c the core server serializes task operations
+                if ( task.status == g_lib.TS_READY ){
+                    g_tasks.taskReady( task._id );
+                } else if ( task.status == g_lib.TS_RUNNING ){
+                    if ( req.queryParams.step != undefined && req.queryParams.step == task.step ){
+                        // This confirms previous step was completed, so update step number
+                        task.step++;
+                        g_db.task.update( task._id, { step: task.step, ut: Math.floor( Date.now()/1000 ) }, { waitForSync: true });
+                    } else if ( req.queryParams.step != undefined && req.queryParams.step >= task.steps ){
+                        throw [ g_lib.ERR_INVALID_PARAM, "Called run on task " + task._id + " with invalid step: " + req.queryParams.step ];
+                    }
+                } else {
+                    throw [ g_lib.ERR_INVALID_PARAM, "Called run on task " + task._id + " with incorrect status: " + task.status ];
+                }
+            }
+        });
+
+        //console.log("task/run - call handler" );
+
+        var result;
+
+        for(;;){
+            try{
+                if ( req.queryParams.err_msg ){
+                    throw [0,req.queryParams.err_msg];
+                }
+
+                result = run_func.call( g_tasks, task );
+                // An empty result means rollback has completed without additional errors
+                if ( !result ){
+                    //console.log("Task run handler stopped rollback" );
+                    result = { cmd: g_lib.TC_STOP, params: g_tasks.taskComplete( task._id, false, task.error )};
+                }
+                break;
+            }catch( e ){
+                var err = Array.isArray(e)?e[1]:e;
+                if ( err.errorMessage )
+                    err = err.errorMessage;
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/ORNL/DataFed2/pullRequest?prid=8791729)
```suggestion
                    {err = err.errorMessage;}
```